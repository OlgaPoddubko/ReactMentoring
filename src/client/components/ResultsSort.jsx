// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { changeSortBy } from '../actions';

type Props = {
  sortBy: string,
  total: number,
  changeSortBy: Function,
  dispatch: Function,
}

export class ResultsSort extends React.Component<Props> {
  constructor(props) {
    super(props);
    (this:any).handleSortByChange = this.handleSortByChange.bind(this);
  }


  handleSortByChange(e) {
    this.props.changeSortBy(e.target.value);
  }

  render() {
    return (
      <div className="results-sort">
        <div className="inner-results-sort">
          <div className="sorting-params">
            <span>sort by</span>
            <button
              className={ this.props.sortBy == 'release_date' ? 'active-sort' : ''}
              type="button"
              value="release_date"
              onClick={this.handleSortByChange}>
              release date
            </button>
            <button
              className={ this.props.sortBy == 'vote_average' ? 'active-sort' : ''}
              type="button"
              value="vote_average"
              onClick={this.handleSortByChange}>
              rating
            </button>
          </div>

          <span className="items-found">{this.props.total} movies found </span>
        </div>
        <style jsx>{`
          .results-sort {
            background: #f5f5f5;
          }
          .inner-results-sort {
            padding: 15px 35px;
            max-width: 1130px;
            margin: 0 auto;
          }
          .sorting-params {
            float: right;
            display: inline-block;
            text-align: right;
          }
          .sorting-params button {
            margin-left: 30px;
            border: none;
            background: #f5f5f5;
            font-size: inherit;
          }
          .active-sort {
            color: #e75f77;
          }
        `}</style>

      </div>
    );
  }
}

const mapStateToProps = state => ({
  sortBy: state.gallery.sortBy,
  total: state.gallery.total,
});

const mapDispatchToProps = dispatch => bindActionCreators({
  changeSortBy
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(ResultsSort);
