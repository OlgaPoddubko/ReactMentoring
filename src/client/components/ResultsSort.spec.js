import React from 'react';
import ResultsSort from './ResultsSort';

describe('render ResultsSort', () => {
  it('should render ResultsSort', () => {
    const wrapper = shallow(<ResultsSort />);
    expect(wrapper).toMatchSnapshot();
  });
});
