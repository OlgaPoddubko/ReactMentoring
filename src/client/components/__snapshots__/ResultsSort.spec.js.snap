// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ResultsSort should render ResultsSort 1`] = `
<div
  className="results-sort"
>
  <div
    className="inner-results-sort"
  >
    <div
      className="sorting-params"
    >
      <span>
        sort by
      </span>
      <button
        className="active-sort"
        onClick={[Function]}
        type="button"
        value="release date"
      >
        release date
      </button>
      <button
        className=""
        onClick={[Function]}
        type="button"
        value="rating"
      >
        rating
      </button>
    </div>
    <span
      className="items-found"
    >
       movies found 
    </span>
  </div>
  <style
    jsx={true}
  >
    
          .results-sort {
            background: #f5f5f5;
          }
          .inner-results-sort {
            padding: 15px 35px;
            max-width: 1130px;
            margin: 0 auto;
          }
          .sorting-params {
            float: right;
            display: inline-block;
            text-align: right;
          }
          .sorting-params button {
            margin-left: 30px;
            border: none;
            background: #f5f5f5;
            font-size: inherit;
          }
          .active-sort {
            color: #e75f77;
          }
        
  </style>
</div>
`;
